### PR TITLE
Cleanup in Surface Detection

### PR DESCRIPTION
**Pull request by Jmeyer1292**
On my way to fixing parameter issues, I've made a lot of changes for cleaning. Should be zero functional changes but it's getting big enough I'd figure I'd make a PR.

### CODE REVIEW

**Comment by woodaaron** [comment]
Your timing utility technically represents a functional change.

File: godel_surface_detection/src/detection/surface_detection.cpp [hunk 1]
```diff
@@ -252,8 +316,43 @@ namespace godel_surface_detection
...
+    {
+      return std::max_element(clouds.begin(), clouds.end(),
+                              [] (const CloudRGB::Ptr& a, const CloudRGB::Ptr& b) {
+        return a->size() < b->size();
+      });
+    }
+
     bool SurfaceDetection::find_surfaces()
     {
+      TimingUtility time_surface_finding (__FUNCTION__); // Poor man's time profiling
```
Comment (woodaaron) [inline godel_surface_detection/src/detection/surface_detection.cpp]: Have you considered __func__ v. __FUNCTION__?

File: godel_surface_detection/src/detection/surface_detection.cpp [hunk 2]
```diff
@@ -252,8 +316,43 @@ namespace godel_surface_detection
     }
 
 
+    template <typename T>
+    static boost::shared_ptr<pcl::PointCloud<T>> downsampleCloud(const pcl::PointCloud<T>& cloud,
+                                                                 std::default_random_engine& random_engine,
+                                                                 int one_in)
+    {
```
Comment (woodaaron) [inline godel_surface_detection/src/detection/surface_detection.cpp:331]: Could use a little documentation here. Even knowing what the function is supposed to accomplish, it took me a little while to understand the "one_in" argument. A brief would make the other comments much more readable/understandable.

**Review by woodaaron — APPROVED** [review]
Minor quibbles.

File: godel_surface_detection/src/detection/surface_detection.cpp (hunk 1 above)
Comment (Jmeyer1292) [reply godel_surface_detection/src/detection/surface_detection.cpp]: I did not know about __func__ till now

File: godel_surface_detection/src/detection/surface_detection.cpp (hunk 2 above)
Comment (Jmeyer1292) [reply godel_surface_detection/src/detection/surface_detection.cpp:331]: +1

**Pull request merged**